django/db/models/fields/json.py

```diff
@@ -0,0 +1,111 @@
+import json
+
+from django import forms
+from django.core import exceptions
+from django.db.utils import NotSupportedError
+from django.utils.translation import gettext_lazy as _
+
+from . import Field
+from .mixins import CheckFieldDefaultMixin
+
+
+def convert_jsonfield_value(value, expression, connection):
+    if connection.vendor == 'oracle':
+        return value.read()
+    return value
```

Comment:
JSON is handle as a `CLOB` on Oracle, hence you can remove `convert_jsonfield_value()` and `JSONField.get_db_converters()` and use the same converter, e.g.
```diff
diff --git a/django/db/backends/oracle/operations.py b/django/db/backends/oracle/operations.py
index 6669ca5beb..6700842bc0 100644
--- a/django/db/backends/oracle/operations.py
+++ b/django/db/backends/oracle/operations.py
@@ -164,7 +164,7 @@ END;
     def get_db_converters(self, expression):
         converters = super().get_db_converters(expression)
         internal_type = expression.output_field.get_internal_type()
-        if internal_type == 'TextField':
+        if internal_type in ['JSONField', 'TextField']:
             converters.append(self.convert_textfield_value)
         elif internal_type == 'BinaryField':
             converters.append(self.convert_binaryfield_value)
```

Reply:
Yes, that's what I meant when I wrote
> "I could've edited the backend, but I decided to override get_db_converters so I don't touch the backend unless it's really necessary."

in the OP.

I didn't touch the backend at all until we decided to add a new database feature. I guess now that I've touched it, might as well do it this way.

Also, does this mean it'd be better if I add the `JSON_VALID` and `IS JSON` constraints in `data_type_check_constraints` in the `DatabaseWrapper` instead of using `db_check` with conditionals in the field?

Reply:
Yes, moving some parts of this implementation to database backends should simplify everything (I'm not sure if check on MySQL is necessary, because it has native `JSON` field).